x/ecocredit/server/marketplace/buy_test.go

```diff
@@ -0,0 +1,338 @@
+package marketplace
+
+import (
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	"google.golang.org/protobuf/types/known/timestamppb"
+	"gotest.tools/v3/assert"
+
+	ecocreditv1 "github.com/regen-network/regen-ledger/api/regen/ecocredit/v1"
+	"github.com/regen-network/regen-ledger/types/math"
+	"github.com/regen-network/regen-ledger/x/ecocredit"
+	"github.com/regen-network/regen-ledger/x/ecocredit/marketplace"
+
+	"github.com/cosmos/cosmos-sdk/orm/types/ormerrors"
+	"github.com/cosmos/cosmos-sdk/testutil/testdata"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+func TestBuy_ValidTradable(t *testing.T) {
```

Comment:
Would really like to see BDD tests... by the time I got to this part of the PR my attention span was like 😴

Reply:
Can you review https://github.com/regen-network/regen-ledger/pull/902 ? I think we need to establish a standard for writing acceptance tests before we try implementing them throughout.

Reply:
That PR looks good to me. Any particular questions you have @ryanchristo ?

Reply:
Not at the moment but I'm sure I will have questions when migrating the existing acceptance tests (#895), especially on how we can reduce complexity and reuse initial setup for multiple test cases.

Reply:
I think we'll want to 1) follow best practices for writing Gherkin - which hopefully that book will help with and 2) come up with best practices for implementing in golang.

I notice you needed to use gomock `AnyTimes`... things like that might be hard to figure out each time and it would be good to document best practices + tips